### ISSUE TITLE
Errors are not returned in promise on form submit

### ISSUE THREAD

**Issue by atschaef** [final-form/final-form#78]
### Are you submitting a **bug report** or a **feature request**?

_Technically_ Feature Request

### What is the current behavior?

When calling submit with an async onSubmit method that returns an error, the errors are not available in the resolved promise.

### What is the expected behavior?

The errors should be returned (if any) when the promise has resolved.

### Sandbox Link

https://codesandbox.io/s/kmll6x2xw7

### What's your environment?

node v8.9.3
npm v5.5.1
react-final-form: 3.1.0
final-form: 4.2.0

### Other information

I believe a simple, yet probably naïve fix, would just to simply return the errors in the 'complete' function of the 'submit' method of FinalForm.js.

https://github.com/final-form/final-form/blob/5c0b10aff85946f1e74a9e90bd04c1fa737892ee/src/FinalForm.js#L627-L649

**Comment by erikras** [comment]
@crorodriguezro To reset after submit, you do this:

```jsx
onSubmit = async (values, form) => {
  const result = await ajax.send(values) // submit the values
  if (result.success) {
    form.reset()
  } else {
    return result.errors
  }
}
```

**Comment by erikras** [comment]
Published fix in [`v4.5.1`](https://github.com/final-form/final-form/releases/tag/v4.5.1).
